src/content/docs/en/reference/adapter-reference.mdx

```diff
@@ -194,18 +194,54 @@ export function start(manifest) {
 
 The following methods are provided:
 
-##### `app.render(request, { routeData, locals })`
+##### `app.render(request: Request, options?: RenderOptions)`
 
 This method calls the Astro page that matches the request, renders it, and returns a Promise to a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) object. This also works for API routes that do not render pages.
 
 ```js
 const response = await app.render(request);
 ```
 
-The method accepts a mandatory `request` argument, and an optional argument for [`routeData`](/en/reference/integrations-reference/#routedata-type-reference) and [`locals`](/en/reference/api-reference/#astrolocals).
+##### `RenderOptions`
+
+The `app.render()` method accepts a mandatory `request` argument, and an optional argument for [`clientAddress`](/en/reference/api-reference/#astroclientaddress),[`locals`](/en/reference/api-reference/#astrolocals), [`routeData`](/en/reference/integrations-reference/#routedata-type-reference).
+
+###### `addCookieHeader`
+
+Whether to automatically add all cookies written by `Astro.cookie.set()` to the response headers.
+When set to `true`, they will be added to the `Set-Cookie` header of the response as comma-separated key=value pairs. You can use the standard `response.headers.getSetCookie()` API to read them individually.
+When set to `false`(default), the cookies will only be available from `App.getSetCookieFromResponse(response)`.
+
+```js
+const response = await app.render(request, { addCookieHeader: true });
+```
+
+###### `clientAddress`
+
+The client IP address that will be made available as `Astro.clientAddress` in pages, and as `ctx.clientAddress` in API routes and middleware.
+
+The example below reads `x-forwarded-for` header and passes it as `clientAddress`, Astro makes this value available to the user as `Astro.clientAddress`.
+```js
+const clientAddress = request.headers.get("x-forwarded-for");
+const response = await app.render(request, { clientAddress });
+```
+
+###### `routeData`
 
 Provide a value for `routeData` if you already know the route to render. Doing so will bypass the internal call to [`app.match`](#appmatchrequest) to determine the route to render.
 
+```js
+const routeData = app.match(request);
+if (routeData) {
+    return app.render(request, { routeData });
+} else {
+    /* adapter-specific 404 response */
+    return new Response(..., { status: 404 });
+}
+```
```

Comment:
``````suggestion
```js {routeData}
const routeData = app.match(request);
if (routeData) {
    return app.render(request, { routeData });
} else {
    /* adapter-specific 404 response */
    return new Response(..., { status: 404 });
}
```
``````

Reply:
Same comment as above! I like the highlighting addition! needs to be `"routeData"` syntax!